api/packages.go

```diff
@@ -223,51 +226,60 @@ func createZipFromFolder(baseFolder string, outFile *os.File) error {
 	w := zip.NewWriter(outFile)
 
 	// Add some files to the archive.
-	addFiles(w, baseFolder, "")
+	err := addFiles(w, baseFolder, "")
+	if err != nil {
+		return err
+	}
 
 	// Make sure to check the error on Close.
-	err := w.Close()
+	err = w.Close()
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
 func isValidFolderOrFile(name string) bool {
-	return strings.HasPrefix(name, "_") || strings.HasPrefix(name, ".")
+	return !strings.HasPrefix(name, "_") && !strings.HasPrefix(name, ".")
 }
 
-func addFiles(w *zip.Writer, basePath string, baseInZip string) {
+func addFiles(w *zip.Writer, basePath string, baseInZip string) error {
 	// Open the Directory
 	files, err := ioutil.ReadDir(basePath)
 	if err != nil {
-		fmt.Println(err)
+		return err
 	}
 
 	for _, file := range files {
-		if isValidFolderOrFile(file.Name()) {
-			return // Ignore unwanted files and directories
+		if !isValidFolderOrFile(file.Name()) {
+			continue
 		}
 
 		if !file.IsDir() {
-			dat, err := ioutil.ReadFile(path.Join(basePath, file.Name()))
+			src, err := os.Open(filepath.Join(basePath, file.Name()))
 			if err != nil {
-				fmt.Println(err)
+				return err
 			}
 
 			// Add some files to the archive.
-			f, err := w.Create(path.Join(baseInZip, file.Name()))
+			dst, err := w.Create(path.Join(baseInZip, file.Name()))
 			if err != nil {
-				fmt.Println(err)
+				return err
```

Comment:
src is not closed in this case. Maybe use defer.

Reply:
Hmm. Problem is that `defer` runs on exit of the function, e.g. not on each iteration of the loop. And I saw an error related to too many open files.

Adding `src.Close()` before returning here.